File: Views/src/components/financial/components/common-components/dto/base-expense-dto.ts
import { MyUser } from "src/components/authentication/dto/my-user";
import { CompanyDto } from "src/shared/entities-dtos/company-dto";
import { BankAccountDto } from "../../bank-account-cards/dto/bank-account-dto";
import { CardDto } from "../../bank-account-cards/dto/card-dto";
import { PixDto } from "../../bank-account-cards/dto/pix-dto";
import { CategoryExpenseDto } from "../../common-components/category-subcategory-expenses/dto/category-expense-dto";
import { SubcategoryExpenseDto } from "../../common-components/category-subcategory-expenses/dto/subcategory-expense-dto";
import { PixExpenseDto } from "../../pix-expenses/dto/pix-expense-dto";
import { RootBase } from "../../../../../shared/entities-dtos/root-base";


export class BaseExpenseDto extends RootBase {
  name :string;
  categoryExpenseId :number;
  categoryExpense :CategoryExpenseDto;
  subcategoryExpenseId :number;
  subcategoryExpense :SubcategoryExpenseDto;
  bankAccountId :number;
  bankAccount :BankAccountDto;
  cardId :number;
  card :CardDto;
  pixExpense:PixExpenseDto;
  pixId :number;
  pix :PixDto;
  price :number;
  interest :number;
  expires :Date;
  wasPaid :Date;
  othersPaymentMethods :string;
  document :string;
  description :string;
  linkCopyBill :string;
  userLinkCopyBill :string;
  passLinkCopyBill :string;
}

File: Application/Services/Operations/Finances/Dtos/InheritanceDto/BaseExpenseDto.cs
using System;

using Application.Services.Operations.Finances.Dtos.Bank;
using Application.Services.Operations.Finances.Dtos.CategorySubcategoryExpenses;
using Application.Services.Operations.Finances.Dtos.PixExpenses;
using Application.Services.Shared.Dtos;


namespace Application.Services.Operations.Finances.Dtos.InheritanceDto
{
    public class BaseExpenseDto : RootBaseDto
    {

        public string Name { get; set; }
        public int CategoryExpenseId { get; set; }
        public CategoryExpenseDto CategoryExpense { get; set; }
        public int SubcategoryExpenseId { get; set; }
        public SubcategoryExpenseDto SubcategoryExpense { get; set; }
        public int? BankAccountId { get; set; }
        public BankAccountDto BankAccount { get; set; }
        public int? CardId { get; set; }
        public CardDto Card { get; set; }
        public int? PixId { get; set; }
        public PixDto Pix { get; set; }
        public PixExpenseDto PixExpense { get; set; }
        public decimal Price { get; set; }
        public decimal Interest { get; set; }
        public DateTime Expires { get; set; }
        public DateTime WasPaid { get; set; }
        public string OthersPaymentMethods { get; set; }
        public string Document { get; set; }
        public string Description { get; set; }
        public string LinkCopyBill { get; set; }
        public string USERLinkCopyBill { get; set; }
        public string PASSLinkCopyBill { get; set; }
    }
}